docs/DevelopersGuide.adoc

```diff
@@ -47,359 +49,1141 @@ the Datomic pull syntax):
+(defresolver `brand-id-from-name
+ {::pc/input #{:product/brand}
+  ::pc/output [:product/brand-id]}
+ (fn [_ {:keys [product/brand]}]
+   {:product/brand-id (get brand->id brand)}))
 
-(def parser
-  ; add our entity reader to our reader list
-  (p/parser {::p/plugins [(p/env-plugin {::p/reader [p/map-reader
-                                                     computed
-                                                     entity]})]}))
+(comment
+  (parser {} [{::latest-product [:product/title :product/brand-id]}])
+  ; => #::{:latest-product #:product{:title "Acoustic Guitar", :brand-id 44151}}
+)
+```
 
-; testing our new queries
-(parser {::db (atom {:characters characters
-                     :tv-shows   tv-shows})}
-        [[:character/id :arya] ; query for the entire entity
-         {[:tv-show/id :rm] ; use the given tv show as the "context" for the join
-          [:tv-show/title
-           {:tv-show/characters [:character/name]}]}])
-; =>
-; {[:character/id :arya] #:character{:name "Arya Stark", :tv-show-id :got}
-;  [:tv-show/id :rm]     #:tv-show{:title      "Rick and Morty"
-;                                  :characters [#:character{:name "Rick Sanshes"}
-;                                               #:character{:name "Summer Smith"}
-;                                               #:character{:name "Morty Smith"}]}}
-----
+Note that our query never said anything about the `:product/brand`. `Connect` automatically walked the path
+`:product/id -> :product/brand -> :product/brand-id` to obtain the information desired by the query!
 
-These building blocks enable quite a bit of query processing. As your graph grows larger it will make sense to use some additional tools to help split your parser into different pieces. In particular `computed` can be written with a dispatch mechanism instead of being represented as a map. See the section about
-<<dispatch-helpers,dispatch helpers>> for more information.
+When a required attribute is not present in the current entity, `Connect` will look for resolvers that can fetch it,
+analyze their inputs, and recursively walk backwards towards the "known data" in the context.
+This is what makes `Connect` powerful: by leveraging the index containing the attribute relationships you can focus on
+writing just the primary `edges` of the graph, and then all possible paths can be walked automatically. You can read
+more about how this works in the Index page.
 
-Here is the complete code for the example:
+In case the path is a dead end (not enough data), `Connect` triggers an error explaining the miss. Let's see that in action:
 
-[source,clojure]
-----
-(ns pathom-docs.hello-entities
-  (:require [com.wsscode.pathom.core :as p]))
+```clojure
+(parser {} [:product/brand])
+; CompilerException clojure.lang.ExceptionInfo: Attribute :product/brand is defined but requirements could not be met. {:attr :product/brand, :entity nil, :requirements (#{:product/id})}
```

Comment:
this is outdated, new versions will just mark this cases as `::p/not-found`, I think we can just remove this part, or replace with a simple note that when not possible it will be a not found

Reply:
I didn't write that part that I remember...perhaps you should fix that?

Reply:
sure, no problem, I can fix this